apps/openmw/mwrender/sky.cpp

```diff
@@ -1337,6 +1345,81 @@ class RainFader : public AlphaFader
     osg::Uniform* mRainIntensityUniform;
 };
 
+class WeatherParticleDrawCallback : public osg::Drawable::DrawCallback
+{
+public:
+    WeatherParticleDrawCallback(osg::Vec3 wrapRange) : osg::Drawable::DrawCallback()
+    {
+        mWrapRange = wrapRange;
+    }
+
+    virtual void drawImplementation(osg::RenderInfo& renderInfo, const osg::Drawable *drawable) const
+    {
+
+        osgParticle::ParticleSystem *ps = (osgParticle::ParticleSystem *) drawable;
+
+        osgParticle::ParticleSystem::ScopedReadLock *lock = new osgParticle::ParticleSystem::ScopedReadLock(*ps->getReadWriteMutex());
+
+        osg::Vec3 cameraPos = renderInfo.getCurrentCamera()->getInverseViewMatrix().getTrans();
+
+        osg::Vec3 cameraOffset = osg::Vec3(
+          fmod(cameraPos.x(), mWrapRange.x()),
+          fmod(cameraPos.y(), mWrapRange.y()),
+          fmod(cameraPos.z(), mWrapRange.z()));
+
+        std::vector<osg::Vec3> positionBackups;
+
+        osg::Matrix toWorld, toLocal;
+
+        toWorld.makeIdentity();
+        toLocal.makeIdentity();
+
+        std::vector<osg::Matrix> worldMatrices = drawable->getWorldMatrices();
+
+        if (!worldMatrices.empty())
```

Comment:
Weird indentation style... is this done on purpose or a case of IDE's gone bad?

Reply:
my bad